cypress/integration/like.spec.ts

```diff
@@ -2,23 +2,20 @@ it('likes an episode then finds it in the library', () => {
   cy.visitAndWaitForHydration('/podcasts/1');
 
   // Find a card and give it a name
-  cy.get('[data-cy="episode-card"]').first().as('episodeCard');
+  cy.findAllByTestId('episode-card').first().as('episodeCard');
 
   // Grab the episode title
   cy.get('@episodeCard')
-    .get('[data-cy="episode-title"]')
-    .first()
+    .findByRole('heading', { level: 2 })
```

Comment:
If you agree with these suggestions, I can implement these changes

Reply:
> it's kind of bleeding through implementation details

That's mostly true, but I couldn't for the life of me figure out how to extract the title in a generic way to use it in queries; the best I could do is the `.invoke('text')` below it.

> I imagine that the user will locate the card they want by the text content of the episode they're looking for instead of relying on the heading structure

I didn't really understand what you meant. Does "text content" mean the episode title or description? Because I can easily imagine users looking at the title first and then the description as a sort of "double check" that it's the correct episode. That's also why I thought the title should be a heading, having in mind screen readers going through them and the users suffering through long descriptions, not realizing they should skip every other paragraph to focus on the titles

Reply:
> I couldn't for the life of me figure out how to extract the title in a generic way

That video I've [linked in the comment above](https://github.com/aabounegm/cast/pull/66#discussion_r815292398) tackles this problem in an alternative way — starting with the data and then proceeding to elements. Basically, we would have a data fixture containing the sample podcasts that we will render and in the test we would iterate over that data and assert it in multiple places.

> Does "text content" mean the episode title or description?

We don't have episode descriptions though. That's why the title will be the one and the only thing that users will read

Reply:
> we would have a data fixture containing the sample podcasts

I thought about that, but I felt that this way we wouldn't care about the data source, and for example it could test Supabase as well when it is implemented. On another thought, you may be right that a fixture is the best solution here (otherwise, we cannot guarantee the existence of data)